### PR TITLE
test: reduce end2end log spam

### PR DESCRIPTION
**Pull request by bradfitz**
Filter expected log output by default, unless a flag is provided, or a
test fails.

This makes it possible to see unexpected things. Having noisy tests
makes it too easy to miss actual problems.

### CODE REVIEW

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19451 (at git rev 4e9bdfe)

File: test/end2end_test.go [hunk 1]
```diff
@@ -1398,3 +1441,98 @@ func leakCheck(t testing.TB) func() {
...
+
+func (lw *lockingWriter) setWriter(w io.Writer) {
+	lw.mu.Lock()
+	defer lw.mu.Unlock()
+	lw.w = w
+}
+
+var testLogOutput = &lockingWriter{w: os.Stderr}
+
+func awaitLogOutput(t *testing.T, maxWait time.Duration, phrase string) {
```
Comment (tamird) [inline test/end2end_test.go]: `t` is unused

**Comment by tamird** [comment]
LGTM.

Leveled logging would be pretty helpful here.

File: test/end2end_test.go (hunk 1 above)
Comment (bradfitz) [inline test/end2end_test.go]: Oh, yeah, I changed my mind about something earlier. Removed.

I have grander plans for all this logging stuff, but this is a baby step.

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19451 (at git rev fa64708)

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19451 (at git rev 35e8ba4)

**Comment by bradfitz** [comment]
Rebased. No new changes.

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19451 (at git rev f8861d0)

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19451 (at git rev 303da9d)

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19451 (at git rev ffa8131)

**Pull request merged**